spring-core/src/test/java/com/baeldung/factorybean/FactoryBeanJavaConfigTest.java

```diff
@@ -16,7 +18,9 @@
 
     @Resource
     private Tool tool;
-    @Resource(name = "&toolFactory")
+
+    @Autowired
+    @Qualifier("&tool")
```

Comment:
Those two lines should be replaced by @Resource("&tool") because we are referencing a one particular resource by its name